### PR TITLE
Add C++ formatting configs

### PR DESCRIPTION
**Pull request by projedi**

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+# Kt N/A.
+NamespaceIndentation: None
+# Kt N/A. The closest is https://kotlinlang.org/docs/reference/coding-conventions.html#class-header-formatting
+ObjCBinPackProtocolList: Never
+# Kt https://kotlinlang.org/docs/reference/coding-conventions.html#formatting
+ObjCBlockIndentWidth: 4
+# Kt https://kotlinlang.org/docs/reference/coding-conventions.html#annotation-formatting
+ObjCSpaceAfterProperty: false
+# Kt https://kotlinlang.org/docs/reference/coding-conventions.html#horizontal-whitespace
+ObjCSpaceBeforeProtocolList: false
```
Comment (SvyatoslavScherbina) [inline .clang-format:141]: Protocols are not like type parameters.
Is it
```obj-c
@interface Foo <Bar>
```
vs
```obj-c
@interface Foo<Bar>
```
?
How do popular style guides handle this?

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+# Kt U. IDEA keeps 1 empty line by default.
+MaxEmptyLinesToKeep: 1
+# Kt N/A.
+NamespaceIndentation: None
+# Kt N/A. The closest is https://kotlinlang.org/docs/reference/coding-conventions.html#class-header-formatting
+ObjCBinPackProtocolList: Never
+# Kt https://kotlinlang.org/docs/reference/coding-conventions.html#formatting
+ObjCBlockIndentWidth: 4
+# Kt https://kotlinlang.org/docs/reference/coding-conventions.html#annotation-formatting
+ObjCSpaceAfterProperty: false
```
Comment (SvyatoslavScherbina) [inline .clang-format:139]: Property attributes don't match annotation arguments semantically. So it makes sense to follow popular style guides here.

File: .clang-format [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+# Kt U. Using true because it looks more like an if-expression with "then" and "else" branches marked "?" and ":"
+BreakBeforeTernaryOperators: true
+# Kt N/A. The closest is https://kotlinlang.org/docs/reference/coding-conventions.html#class-header-formatting AfterColon looks the most consistent
+BreakConstructorInitializers: AfterColon
+# Kt https://kotlinlang.org/docs/reference/coding-conventions.html#class-header-formatting
+BreakInheritanceList: AfterColon
+# Kt U. Choosing to break long strings to fit into line length
+BreakStringLiterals: true
+# Kt U. IDEA displays a vertical guide at 120. Choosing it to avoid very long lines.
+ColumnLimit: 140
```
Comment (SvyatoslavScherbina) [inline .clang-format:92]: Can this cause clang-format to merge multiline statement to single line?

File: codestyle/cpp/CLionFormat.xml [hunk 4]
```diff
@@ -0,0 +1,58 @@
+<code_scheme name="Project" version="173">
```
Comment (SvyatoslavScherbina) [inline codestyle/cpp/CLionFormat.xml:1]: I'm getting weird formatting for 
```c++
  switch (container->tag()) {
    case CONTAINER_TAG_STACK:
      break;
    case CONTAINER_TAG_LOCAL:
      incrementRC</* Atomic = */ false>(container);
      break;
    /* case CONTAINER_TAG_FROZEN: case CONTAINER_TAG_SHARED: */
    default:
      incrementRC</* Atomic = */ true>(container);
      break;
  }
```
->
```c++
    switch (container->tag()) {
        case CONTAINER_TAG_STACK:break;
        case CONTAINER_TAG_LOCAL:incrementRC</* Atomic = */ false>(container);
            break;
            /* case CONTAINER_TAG_FROZEN: case CONTAINER_TAG_SHARED: */
        default:incrementRC</* Atomic = */ true>(container);
            break;
    }
```

Comment (projedi) [reply codestyle/cpp/CLionFormat.xml:1]: That's "keep simple cases in one line" in "'switch' statement" option firing. Unfortunately, this particular option doesn't seem to get serialized to the xml file.

File: .clang-format (hunk 3 above)
Comment (projedi) [reply .clang-format:92]: Yes. See `AllowShort…OnASingleLine`

File: .clang-format (hunk 1 above)
Comment (projedi) [reply .clang-format:141]: For this comment and the one below it:
```
> clang-format -style=LLVM -dump-config | grep ObjCSpace
ObjCSpaceAfterProperty: false
ObjCSpaceBeforeProtocolList: true
> clang-format -style=Google -dump-config | grep ObjCSpace
ObjCSpaceAfterProperty: false
ObjCSpaceBeforeProtocolList: true
> clang-format -style=Mozilla -dump-config | grep ObjCSpace
ObjCSpaceAfterProperty: true
ObjCSpaceBeforeProtocolList: false
> clang-format -style=WebKit -dump-config | grep ObjCSpace
ObjCSpaceAfterProperty: true
ObjCSpaceBeforeProtocolList: true
```

**Review by SvyatoslavScherbina — APPROVED** [review]

**Pull request merged**